simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -283,20 +324,35 @@ bool SwitchMmu::CheckIngressAdmission(uint32_t port, uint32_t qIndex, uint32_t p
 
 	switch (type) {
 	case LOSSY:
-		if ( (psize + ingress_bytes[port][qIndex] > Threshold(port, qIndex, "ingress", type) && psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex])
-		        // || (psize + ingressPoolUsed > ingressPool )
-		        || (psize + totalSharedUsed > bufferPool) ) {
+		        // if ingress bytes is greater than the ingress threshold 
+		if ( (psize + ingress_bytes[port][qIndex] > Threshold(port, qIndex, "ingress", type)
+				// AND if the reserved is usedup
+		        && psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex]) 
+		        /*if the ingress pool is full. With DT, this condition is redundant. 
+		        	This is just to account for any badly configured buffer or buffer sharing if any.*/
+		        || (psize + (totalUsed-xoffTotalUsed) > ingressPool) 
+		        // or if the switch buffer is full
+		        || (psize + totalUsed > bufferPool) )  
+		{
 			return false;
 		}
 		else {
 			return true;
 		}
 		break;
 	case LOSSLESS:
-		if ( ( (psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex])
-		        && (psize + ingress_bytes[port][qIndex] > Threshold(port, qIndex, "ingress", type) + xoff[port][qIndex]) )
-		        || (psize + ingressPoolUsed > ingressPool + xoffTotal)
-		        || (psize + totalSharedUsed > bufferPool)  ) {
+				// if reserved is used up
+		if ( ( (psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex]) 
```

Comment:
<code>psize + ingress_bytes[port][qIndex] > reserveIngress[port][qIndex]</code> seems redundant to me